ui/src/components/role/RoleSectionRow.js

```diff
@@ -0,0 +1,386 @@
+    padding: 5px 0 5px 0;
+    vertical-align: middle;
+    word-break: break-all;
+    width: 6%;
+`;
+
+const TrStyled = styled.tr`
+    background-color: ${(props) => props.color};
+`;
+
+const MenuDiv = styled.div`
+    padding: 5px 10px;
+    background-color: black;
+    color: white;
+    font-size: 12px;
+`;
+
+const LeftSpan = styled.span`
+    padding-left: 20px;
+`;
+
+export default class RoleSectionRow extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = this.props.api;
+        this.onSubmitDelete = this.onSubmitDelete.bind(this);
+        this.onClickDeleteCancel = this.onClickDeleteCancel.bind(this);
+        this.saveJustification = this.saveJustification.bind(this);
+        this.state = {
+            name: NameUtils.getShortName(':role.', this.props.details.name),
+            showDelete: false,
+        };
+        this.localDate = new DateUtils();
+    }
+
+    saveJustification(val) {
+        this.setState({ deleteJustification: val });
+    }
+
+    onClickDelete(name) {
+        this.setState({
+            showDelete: true,
+            deleteName: name,
+        });
+    }
+
+    onClickFunction(route, domain, role) {
+        Router.pushRoute(route, { domain: domain, role: role });
+    }
+
+    onSubmitDelete(domain) {
+        let roleName = this.state.deleteName;
+        if (
+            this.props.justificationRequired &&
+            (this.state.deleteJustification === undefined ||
+                this.state.deleteJustification.trim() === '')
+        ) {
+            this.setState({
+                errorMessage: 'Justification is required to delete a role',
+            });
+            return;
+        }
+
+        this.api
+            .deleteRole(
+                domain,
+                roleName,
+                this.state.deleteJustification
+                    ? this.state.deleteJustification
+                    : 'deleted using Athenz UI',
+                this.props._csrf
+            )
+            .then(() => {
+                this.setState({
+                    showDelete: false,
+                    deleteName: null,
+                    deleteJustification: null,
+                    errorMessage: null,
+                });
+                this.props.onUpdateSuccess(
+                    `Successfully deleted role ${roleName}`
+                );
+            })
+            .catch((err) => {
+                this.setState({
+                    errorMessage: RequestUtils.xhrErrorCheckHelper(err),
+                });
+            });
+    }
+
+    onClickDeleteCancel() {
+        this.setState({
+            showDelete: false,
+            deleteName: '',
+        });
+    }
+
+    render() {
+        let rows = [];
+        let left = 'left';
+        let center = 'center';
+        let role = this.props.details;
+        let color = this.props.color;
+        let idx = this.props.idx;
+
+        let clickMembers = this.onClickFunction.bind(
```

Comment:
let's move the below function variables to the constructor, as on every re-render these will be created again causing the unnecessary function creations

Reply:
but there is a parameter for state